libs/failsafe-controller/src/main/java/net/jodah/failsafe/internal/actions/ActionList.java

```diff
@@ -426,7 +660,24 @@ public Stubber0 waitTo(String condition) {
     }
 
     /**
-     * Indicates to block failsafe the next time it attempts a retry until its execution is
+     * Indicates to block failsafe the next time it attempts a retry until the specified
+     * condition/latch is notified. After the specified condition/latch has been notified, the
+     * controller will move on to the next action and execute it right away.
+     *
+     * <p>If the specified condition/latch has already been notified then the controller will move
+     * on to the next action.
+     *
+     * @param condition the condition/latch to wait for
+     * @return a stub construct for chaining
+     * @throws IllegalArgumentException if <code>condition</code> is <code>null</code>
+     */
+    public Stubber0 waitingTo(String condition) {
+      add(expectation -> new WaitForAction<>(expectation, "waitingTo", condition));
```

Comment:
✏️  Just a nitpick, but after seeing this it seems like naming it `WaitAction` would be a little more generic since it handles both `waitTo` and `waitFor`

Reply:
`waitTo()` was added as an alias for `waitFor()` because in some cases it reads better in test code. I do have other aliases like that that are completely different. For example, `doNothing()`, `doReturn()`, and `returning()`. It won't always be black or white. As such, I am going to keep it as is for now.